runtime/vm/JVMImage.cpp

```diff
@@ -88,52 +87,172 @@ JVMImage::getInstance()
 	return _jvmInstance;
 }
 
-void * 
+//TODO: This will be expanded once warm run set up is finished
+ImageRC
+JVMImage::setupWarmRun()
+{
+	ImageRC result = IMAGE_OK;
+
+	if (!readImageFromFile()) {
+		result = IMAGE_ERROR;
+	}
+
+	return result;
+}
+
+ImageRC
+JVMImage::setupColdRun()
+{
+	ImageRC result = IMAGE_OK;
+
+	if (allocateImageMemory(INITIAL_IMAGE_SIZE) == NULL) {
+		result = IMAGE_ERROR;
```

Comment:
If allocateImageMemory fails, will it end up causing a seg fault for the other function calls? If so we should return right away